### PR TITLE
fix: forcing make_image_grid to generate square sprite image

### PR DESCRIPTION
**Pull request by YusongLeng**
*Issues, if available:* #35 

*Description of changes:* one line changed in utils/make_image_grid to make sure the produced sprite image is precisely square

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: python/mxboard/utils.py [hunk 1]
```diff
@@ -153,7 +153,7 @@ def norm_range_helper(t, val_range):
     # make the batch of images into a grid
     nmaps = tensor.shape[0]
     xmaps = min(nrow, nmaps)
-    ymaps = int(np.ceil(float(nmaps) / xmaps))
+    ymaps = xmaps
```
Comment (reminisce) [inline python/mxboard/utils.py]: This function is also called by `add_image` in addition to `add_embedding`, and we don't want to change the behavior of `add_image`. Could you add a parameter called `square_image=False` to `make_image_grid` and its callers, and only use `square_image=True` when called from `add_embedding`?

Comment (YusongLeng) [reply python/mxboard/utils.py]: Hi reminisce,

Many thanks for your notification. I'll edit it according to your suggestion and get it back to you soon!

**Comment by reminisce** [comment]
Thanks for the contribution. Could you confirm that you have
1. `make pylint` and it gives zero errors.
2. run all the unit tests in `tests/python`?

**Comment by YusongLeng** [comment]
Hi reminisce,

Thanks for the notes! I run the given `run_pylint.sh` and got the following results:

```
************* Module mxboard.record_writer
python/mxboard/record_writer.py:25:0: R0205: Class 'RecordWriter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
************* Module mxboard.event_file_writer
python/mxboard/event_file_writer.py:38:0: R0205: Class 'EventsWriter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
python/mxboard/event_file_writer.py:115:0: R0205: Class 'EventFileWriter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
************* Module mxboard.writer
python/mxboard/writer.py:38:0: R0205: Class 'SummaryToEventTransformer' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
python/mxboard/writer.py:187:0: R0205: Class 'SummaryWriter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
************* Module mxboard.utils
python/mxboard/utils.py:219:0: C0301: Line too long (111/100) (line-too-long)
python/mxboard/utils.py:275:0: C0301: Line too long (103/100) (line-too-long)

------------------------------------------------------------------
Your code has been rated at 9.90/10 (previous run: 4.68/10, +5.22)
```

Also, by running `tests/python/unittest/test_logging.py`, I got:

```
[INFO] Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=1397186505 to reproduce.
................
----------------------------------------------------------------------
Ran 16 tests in 0.152s

OK
```

And, `tests/python/gpu/test_logging_gpu.py` gave me:

```
[INFO] Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=828935441 to reproduce.
................
----------------------------------------------------------------------
Ran 16 tests in 1.393s

OK
```

So, I think the changes have passed the test, right? Please correct me if I'm wrong. Thanks!

**Comment by reminisce** [comment]
Could you fix the following lint errors? The number of characters of a line cannot be greater than 100.

```
************* Module mxboard.utils
python/mxboard/utils.py:219:0: C0301: Line too long (111/100) (line-too-long)
python/mxboard/utils.py:275:0: C0301: Line too long (103/100) (line-too-long)
```

File: python/mxboard/utils.py [hunk 2]
```diff
@@ -211,7 +216,7 @@ def _prepare_image(img, nrow=8, padding=2):
     assert img.ndim == 2 or img.ndim == 3 or img.ndim == 4
 
     if img.dtype == np.uint8:
-        return make_image_grid(img, nrow=nrow, padding=padding).transpose((1, 2, 0))
+        return make_image_grid(img, nrow=nrow, padding=padding, square_image=square_image).transpose((1, 2, 0))
```
Comment (reminisce) [inline python/mxboard/utils.py]: lint: Line too long (111/100)

File: python/mxboard/utils.py [hunk 3]
```diff
@@ -267,7 +272,7 @@ def _make_sprite_image(images, save_path):
     assert isinstance(images, NDArray)
     shape = images.shape
     nrow = int(np.ceil(np.sqrt(shape[0])))
-    _save_image(images, os.path.join(save_path, 'sprite.png'), nrow=nrow, padding=0)
+    _save_image(images, os.path.join(save_path, 'sprite.png'), nrow=nrow, padding=0, square_image=True)
```
Comment (reminisce) [inline python/mxboard/utils.py]: lint: Line too long (103/100)

**Comment by YusongLeng** [comment]
Hi reminisce,

Issue fixed, and the code passed the pylint and tests without errors. Now, I think it's ready to go~ :)

```
************* Module mxboard.record_writer
python/mxboard/record_writer.py:25:0: R0205: Class 'RecordWriter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
************* Module mxboard.event_file_writer
python/mxboard/event_file_writer.py:38:0: R0205: Class 'EventsWriter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
python/mxboard/event_file_writer.py:115:0: R0205: Class 'EventFileWriter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
************* Module mxboard.writer
python/mxboard/writer.py:38:0: R0205: Class 'SummaryToEventTransformer' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
python/mxboard/writer.py:187:0: R0205: Class 'SummaryWriter' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)

------------------------------------------------------------------
Your code has been rated at 9.93/10 (previous run: 9.90/10, +0.03)
```

**Pull request merged**